coverage/diff/view.go

```diff
@@ -0,0 +1,76 @@
+package diff
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/test-infra/gopherage/pkg/cov/junit/calculation"
+)
+
+// formatPercentage converts a coverage ratio into a string value to be displayed by coverage robot
+func formatPercentage(ratio float32) string {
+	if ratio < 0 {
+		return "Do not exist"
+	}
+	return fmt.Sprintf("%.1f%%", ratio*100)
+}
+
+// deltaDisplayed converts a coverage ratio delta into a string value to be displayed by coverage robot
+func deltaDisplayed(change *coverageChange) string {
+	if change.baseRatio < 0 {
+		return ""
+	}
+	return fmt.Sprintf("%.1f", (change.newRatio-change.baseRatio)*100)
+}
+
+// makeTable checks each coverage change and produce the table content for coverage bot post
+// It also report on whether any coverage fells below the given threshold
+func makeTable(changes []*coverageChange, coverageThreshold float32) (string, bool) {
+	var rows []string
+	isCoverageLow := false
+	for _, change := range changes {
+		filePath := change.name
+		rows = append(rows, fmt.Sprintf("%s | %s | %s | %s",
+			filePath,
+			formatPercentage(change.baseRatio),
+			formatPercentage(change.newRatio),
+			deltaDisplayed(change)))
+
+		if change.newRatio < coverageThreshold {
+			isCoverageLow = true
+		}
+	}
+	return strings.Join(rows, "\n"), isCoverageLow
+}
+
+// ContentForGithubPost constructs the message covbot posts
+func ContentForGithubPost(baseCovList, newCovList *calculation.CoverageList, jobName string, coverageThreshold float32) (
```

Comment:
This should take a pair of `[]*cover.Profile`s, to avoid exposing the dependency on `calculation` to its caller.